atd-vze/src/views/Crashes/CrashChangeLog.js

```diff
@@ -0,0 +1,216 @@
+        return a.getDate() + ' ' + months[a.getMonth()] + ' ' + a.getFullYear() + ' ' + a.getHours() + ':' + a.getMinutes();
+    }
+
+    /**
+     * Shows the modal box by changing the state, does not alter the body.
+     */
+    showModal() {
+        this.setState({
+            modal: true
+        });
+    }
+
+    /**
+     * Closes the modal box by changing the state and emptying the modal body.
+     */
+    closeModal() {
+        this.setState({
+            modal: false,
+            modalBody: null
+        });
+    }
+
+    /**
+     * Compares the archived json object against the current data object as populated via props.
+     * @param {Object} record's json
+     */
+    compare(record) {
+        // Holds only the keys
+        let diff = [];
+
+        // Holds the html for our modal box
+        let modalBody = null;
+
+        // Iterate through our record's key & values
+        for (let [key, value] of Object.entries(this.props.data.atd_txdot_crashes[0])) {
+            // Let's get rid of typename
+            if(key === "__typename") continue;
+
+            try {
+                // Gather the archived value for the current field
+                let archivedRecordValue = record.record_json[key];
+
+                // If the value is different from the current value, then put it in the diff array.
+                if(archivedRecordValue !== value) {
+                    diff.push({
+                        'original_record_key': key,
+                        'original_record_value': value,
+                        'archived_record_value': archivedRecordValue
+                    });
+                }
+            } catch (error) {
+                console.log(error);
+            }
+        }
+
+        // For each entry created in the diff array, generate an HTML table row.
+        let modalItems = diff.map(item => (
+            <tr key={record.id}>
+                <td>
+                    {item.original_record_key}
+                </td>
+                <td>
+                    <Badge color="primary">{String(item.original_record_value)}</Badge>
+                </td>
+                <td>
+                    <Badge color="danger">{String(item.archived_record_value)}</Badge>
+                </td>
+            </tr>
+        ));
+
+        // Generate the body of the modal box
+        modalBody = <section>
+            <h6>Crash ID: <Badge color={"secondary"} className={"float-right"}>{record.record_crash_id}</Badge></h6>
+            <h6>Archive Date: <Badge color={"secondary"} className={"float-right"}>{record.update_timestamp}</Badge></h6>
+            <h6>Created by: <Badge color={"secondary"} className={"float-right"}>{record.record_json.updated_by}</Badge></h6>
+            &nbsp;
+            <Table responsive>
+                <thead>
+                <tr>
+                    <td>Field</td>
+                    <td>Current Value</td>
+                    <td>Archived Value</td>
```

Comment:
Let's call this "Previous Value" for simplicity

Reply:
done! 👍